types/whatwg-streams/index.d.ts

```diff
@@ -1,25 +1,27 @@
 // Type definitions for Streams API
 // Project: https://github.com/whatwg/streams
 // Definitions by: Kagami Sascha Rosylight <https://github.com/saschanaz>
+// Definitions by: Konstantin Simon Maria Möllers <https://github.com/ksm2>
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// TypeScript Version: 2.3
 
-export interface ReadableStreamSource {
-    start?(controller: ReadableStreamDefaultController): void | Promise<void>;
-    pull?(controller: ReadableStreamDefaultController): void | Promise<void>;
+export interface ReadableStreamSource<R = ArrayBufferView> {
```

Comment:
`R extends ArrayBufferView = ArrayBufferView`?

Reply:
Actually, by the Spec, [everything can be a chunk type](https://streams.spec.whatwg.org/#model), so `R` does not have to extend `ArrayBufferView`.